tests/random_test.py

```diff
@@ -336,6 +336,27 @@ def testGammaGradType(self):
     # Should not crash with a type error.
     api.vjp(f, a, b)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_lam={}_{}".format(lam, dtype),
+       "lam": lam, "dtype": onp.dtype(dtype).name}
+      for lam in [5.0, 50.0, 5e6]
+      for dtype in [onp.int32, onp.int64]))
+  def testPoisson(self, lam, dtype):
+    key = random.PRNGKey(0)
+    rand = lambda key, lam: random.poisson(key, lam, (10000,), dtype)
+    crand = api.jit(rand)
+
+    uncompiled_samples = rand(key, lam)
+    compiled_samples = crand(key, lam)
+
+    for samples in [uncompiled_samples, compiled_samples]:
+      self._CheckChiSquared(samples, scipy.stats.poisson(lam).pmf)
```

Comment:
Note: I don't have much faith in this statistical test. It still passes if I add 100 to all samples, which is disconcerting.

Either this is a bug or we should add some other checks, e.g., statistics of the distribution.

Reply:
This is fixed by https://github.com/google/jax/pull/2847, which I've also merged into this PR.